com.solvespace.SolveSpace.json

```diff
@@ -242,19 +242,22 @@
             ],
             "sources": [
                 {
-                    "type": "git",
-                    "url": "https://github.com/solvespace/solvespace.git",
-                    "tag": "v3.1",
-                    "commit": "70bde63cb32a7f049fa56cbdf924e2695fcb2916",
-                    "disable-shallow-clone": true,
+                    "type": "archive",
+                    "url": "https://github.com/solvespace/solvespace/releases/download/v3.1/solvespace-3.1.tar.xz",
+                    "sha256": "34a273ce642d0c77b8f101463730ed4eade7d90cfabfe486f3e7cbf494ff132a",
                     "x-checker-data": {
                         "is-main-source": true,
                         "type": "json",
                         "url": "https://api.github.com/repos/solvespace/solvespace/releases/latest",
-                        "tag-query": ".tag_name",
-                        "version-query": "$tag | sub(\"^v\"; \"\")",
-                        "timestamp-query": ".published_at"
+                        "version-query": ".tag_name | sub(\"^v\"; \"\")",
+                        "url-query": ".assets[] | select(.name==\"solvespace-\" + $version + \".tar.xz\") | .browser_download_url"
                     }
+                },
+                {
+                    "type": "shell",
+                    "commands": [
+                      "sed -i '/^include(GetGitCommitHash)/d' CMakeLists.txt"
```

Comment:
Better to actually put the commit/tag in there as requested by the comment.

Reply:
We can't automate this commit hash update with the available Flatpak packaging tools, so switch back to get or fix the tarball. Possibly

```
diff --git a/.github/workflows/source-tarball.yml b/.github/workflows/source-tarball.yml
index 8bc0622..3b927a1 100644
--- a/.github/workflows/source-tarball.yml
+++ b/.github/workflows/source-tarball.yml
@@ -21,7 +21,11 @@ jobs:
         dir_name="solvespace-${version}"
         archive_name="${dir_name}.tar.xz"
         archive_path="${HOME}/${archive_name}"
+        commit_sha="$GITHUB_SHA"

+        sed -e 's/^\(include(GetGitCommitHash)\)/#\1/' \
+            -e 's/^# \(set(GIT_COMMIT_HASH\).*/\1 '"$commit_sha"')/' \
+            -i CMakeList.txt
         echo "::set-output name=archive_name::${archive_name}"
         echo "::set-output name=archive_path::${archive_path}"
```

Reply:
Ah, the latter is a good idea for future releases. For now perhaps just put "flatpak" as the hash?

Reply:
OK, submitted this workflow change upstream. https://github.com/solvespace/solvespace/pull/1265

Reply:
Done! And it still builds correctly.